### PR TITLE
refactor: misc fixes logging, annotations

### PR DESCRIPTION
**Pull request by anshbansal**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-service/restli-client/src/main/java/com/linkedin/common/client/BaseClient.java [hunk 1]
```diff
@@ -41,8 +42,10 @@ protected <T> Response<T> sendClientRequest(final AbstractRequestBuilder<?, ?, ?
   @SneakyThrows
   protected <T> Response<T> sendClientRequest(
       final AbstractRequestBuilder<?, ?, ? extends Request<T>> requestBuilder,
-      @Nonnull final Authentication authentication) throws RemoteInvocationException {
-    requestBuilder.addHeader(HttpHeaders.AUTHORIZATION, authentication.getCredentials());
+      @Nullable final Authentication authentication) throws RemoteInvocationException {
+    if (authentication != null) {
```
Comment (anshbansal) [inline metadata-service/restli-client/src/main/java/com/linkedin/common/client/BaseClient.java:46]: There is a method just above this which passes null as the second argument.

File: metadata-service/restli-client/src/main/java/com/linkedin/common/client/BaseClient.java [hunk 2]
```diff
@@ -66,8 +69,6 @@ protected <T> Response<T> sendClientRequest(
 
   @Override
   public void close() {
-    if (_client != null) {
-      _client.shutdown(new FutureCallback<>());
-    }
+    _client.shutdown(new FutureCallback<>());
```
Comment (anshbansal) [inline metadata-service/restli-client/src/main/java/com/linkedin/common/client/BaseClient.java:72]: the client is created in init method with `Objects.requireNonNull` so this cannot be null

File: metadata-service/restli-client/src/main/java/com/linkedin/entity/client/EntityClient.java [hunk 3]
```diff
@@ -359,7 +359,7 @@ public VersionedAspect getAspectOrNull(@Nonnull String urn, @Nonnull String aspe
 
   public List<EnvelopedAspect> getTimeseriesAspectValues(@Nonnull String urn, @Nonnull String entity,
       @Nonnull String aspect, @Nullable Long startTimeMillis, @Nullable Long endTimeMillis, @Nullable Integer limit,
-      @Nonnull Boolean getLatestValue, @Nullable Filter filter, @Nonnull Authentication authentication)
+      @Nullable Boolean getLatestValue, @Nullable Filter filter, @Nonnull Authentication authentication)
```
Comment (anshbansal) [inline metadata-service/restli-client/src/main/java/com/linkedin/entity/client/EntityClient.java:362]: Few places in code which pass this as null

File: datahub-upgrade/src/main/java/com/linkedin/datahub/upgrade/nocode/DataMigrationStep.java [hunk 4]
```diff
@@ -95,7 +95,7 @@ public Function<UpgradeContext, UpgradeStepResult> executable() {
           try {
             urn = Urn.createFromString(oldAspect.getKey().getUrn());
           } catch (Exception e) {
-            throw new RuntimeException(String.format("Failed to bind Urn with value %s into Urn object", oldAspect.getKey().getUrn(), e));
+            throw new RuntimeException(String.format("Failed to bind Urn with value %s into Urn object: %s", oldAspect.getKey().getUrn(), e));
```
Comment (jjoyce0510) [inline datahub-upgrade/src/main/java/com/linkedin/datahub/upgrade/nocode/DataMigrationStep.java:98]: please remove this from here and change to this so we get full trace:

```
throw new RuntimeException(String.format("Failed to bind Urn with value %s into Urn object", oldAspect.getKey().getUrn()), e);
```

File: metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java [hunk 5]
```diff
@@ -666,7 +666,7 @@ protected List<Pair<String, UpdateAspectResult>> ingestAspectsToLocalDB(
   }
 
   @Nonnull
-  protected SystemMetadata generateSystemMetadataIfEmpty(SystemMetadata systemMetadata) {
+  protected SystemMetadata generateSystemMetadataIfEmpty(@Nullable SystemMetadata systemMetadata) {
```
Comment (jjoyce0510) [inline metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java:669]: thanks!

File: metadata-service/restli-client/src/main/java/com/linkedin/common/client/BaseClient.java (hunk 1 above)
Comment (jjoyce0510) [reply metadata-service/restli-client/src/main/java/com/linkedin/common/client/BaseClient.java:46]: thanks for fixing this

File: metadata-service/restli-client/src/main/java/com/linkedin/common/client/BaseClient.java (hunk 2 above)
Comment (jjoyce0510) [reply metadata-service/restli-client/src/main/java/com/linkedin/common/client/BaseClient.java:72]: thank you!

**Review by jjoyce0510 — APPROVED** [review]
LGTM. Thank you!

File: datahub-upgrade/src/main/java/com/linkedin/datahub/upgrade/nocode/DataMigrationStep.java (hunk 4 above)
Comment (anshbansal) [reply datahub-upgrade/src/main/java/com/linkedin/datahub/upgrade/nocode/DataMigrationStep.java:98]: Done as part of https://github.com/datahub-project/datahub/pull/7606

**Pull request merged**